src/components/DayPickerRangeController.jsx

```diff
@@ -512,6 +514,8 @@ export default class DayPickerRangeController extends React.PureComponent {
         }
       }
 
+      onDatesChange({ startDate, endDate });
```

Comment:
i'm a little worried about this line being the first of twice - can we add a test with spies, for each of these branches, to ensure it's called both once, and in the proper order?

Reply:
@ljharb Can you explain what you mean by "this line being the first of twice"? 
`onDatesChange` will only be called once in this function, and I can add a test to ensure this

Reply:
That's my exact concern; a test is great :-)